### PR TITLE
[Android]-Custom CommissioningFlow - Redirect back to ChipTool after commissioning 

### PR DESCRIPTION
**Pull request by lijujayakumar**
Custom CommissioningFlow - Redirect back to ChipTool after commissioning 

#### Change overview
This PR defines a URI to the android app, such that 3P apps can redirect back the users to ChipTool as part of Custom CommissioningFlow Implementation.

#### Testing
How was this tested? (at least one bullet point required)
* If manually tested, using the following URI - mt://modelinfo?payload=eyJjb21taXNzaW9uaW5nRmxvdyI6MCwiZGlzY292ZXJ5Q2FwYWJpbGl0aWVzIjpbIkJMRSJdLCJk aXNjcmltaW5hdG9yIjozODQwLCJvcHRpb25hbFFyQ29kZUluZm9NYXAiOnt9LCJwcm9kdWN0SWQi OjY1Mjc5LCJzZXR1cFBpbkNvZGUiOjIwMjAyMDIxLCJ2ZW5kb3JJZCI6OTA1MCwidmVyc2lvbiI6 MH0=

Screenshots
* Landing page when redirected using the URI
Landing
* Device Info page (Payload parsed from the URI)
DeviceInfo

### CODE REVIEW

**Review by chrisdecenzo — APPROVED** [review]

**Review by chrisbeach59 — APPROVED** [review]

**Comment by andy31415** [comment]
@austinh0 @xylophone21 ?

File: src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/CHIPToolActivity.kt [hunk 1]
```diff
@@ -72,6 +74,10 @@ class CHIPToolActivity :
 
     if (intent?.action == NfcAdapter.ACTION_NDEF_DISCOVERED)
       onNfcIntent(intent)
+
+    if(Intent.ACTION_VIEW == intent?.action) {
```
Comment (austinh0) [inline src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/CHIPToolActivity.kt:78]:
```suggestion
    if (Intent.ACTION_VIEW == intent?.action) {
```

File: src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/CHIPToolActivity.kt [hunk 2]
```diff
@@ -222,6 +228,54 @@ class CHIPToolActivity :
         .show()
   }
 
+  private fun onReturnIntent(intent: Intent) {
+    val appLinkData = intent.data
+    // Require URI schema "mt:"
+    if (!appLinkData?.scheme.equals("mt")) return
+    // Require URI host "modelinfo"
+    if (!appLinkData?.host.equals("modelinfo")) return
```
Comment (austinh0) [inline src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/CHIPToolActivity.kt:236]: Might be more clear to log some errors in these cases

**Review by austinh0 — APPROVED** [review]

File: src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/CHIPToolActivity.kt (hunk 2 above)
Comment (lijujayakumar) [reply src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/CHIPToolActivity.kt]: Detailed logging added

**Review by woody-apple — APPROVED** [review]

**Pull request merged**